target/linux/ath79/image/nand.mk

```diff
@@ -77,6 +90,43 @@ define Device/domywifi_dw33d
 endef
 TARGET_DEVICES += domywifi_dw33d
 
+define Device/dongwon_dw02-412h
+  SOC := qca9557
+  DEVICE_VENDOR := Dongwon T&I
+  DEVICE_MODEL := DW02-412H
+  DEVICE_ALT0_VENDOR := T&Ie
+  DEVICE_ALT0_MODEL := DW02-412H
+  DEVICE_ALT1_VENDOR := KT
+  DEVICE_ALT1_MODEL := GiGA WiFi home
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-storage kmod-ath10k-ct \
+	ath10k-firmware-qca988x-ct
+  KERNEL_SIZE := 16384k
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma | kt-headers
+  KERNEL_INITRAMFS := kernel-bin | append-dtb | lzma | uImage lzma | kt-headers
+  UBINIZE_OPTS := -E 5
+  IMAGES += factory.img
+  IMAGE/factory.img := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi | \
+	check-size
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata | \
+	check-size
```

Comment:
`check-size` compares the image size with `IMAGE_SIZE`, and for sysupgrade.bin comparing a compressed image size with an (uncompressed) flash partition size is not really fair. You can just skip it.